File: scrb-introtask/intro_py_pkg/intro_py_pkg/publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

#tried String and it works, but i want to try with other types.
from std_msgs.msg import Int8


class IntroPublisher(Node):
    def __init__(self):
        super().__init__("intro_publisher") #constructor initializes with name
        
        #recall that a topic is like a bus that carries a message of some type.
        #in this case we use the String type from std_msgs as our message.
        # we set our buffer (queue size) to hold 5 messages.
         
        # Publisher for String messages on 'string_topic'
        self.string_publisher_ = self.create_publisher(String, 'string_topic', 5)
        # Publisher for Int8 messages on 'int_topic'
        self.int8_publisher_ = self.create_publisher(Int8, 'int_topic', 5)
        self.count = 0
        self.timer = self.create_timer(1.0, self.timer_callback) # starts a timer that calls the timer_callback function every second.
        
    def timer_callback(self):
        # create and publish Int8 message
        int8msg = Int8()
        int8msg.data = self.count
        self.int8_publisher_.publish(int8msg)
        self.get_logger().info(f'Publishing on int_topic: x={int8msg.data}')

        # create and publish String message
        strmsg = String()
        strmsg.data = f'Hello {self.count}'
        self.string_publisher_.publish(strmsg)
        self.get_logger().info(f'Publishing on string_topic: {strmsg.data}')

        # Increment counter for next message
        self.count += 1
        
        
        
        
        

def main(args=None):
    rclpy.init(args=args)
    node = IntroPublisher() 
    rclpy.spin(node)
    
    node.destroy_node() 
    rclpy.shutdown()



if __name__ == "__main__":
	main()

File: scrb-introtask/my_cpp_pkg/src/publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
//base package and msg string type

class IntroPublisher : public rclcpp::Node 
{

    
	public:
		//the declaration of count_(0) declares together WITH the constructor instead of within it.
		IntroPublisher() : Node("intro_publisher"),count_(0) {//constructor
			//this-> is the equivalent of using self.
			//<std_msg...> defines the type	
			//we use the string_topic topic and set the queue/buffer to 5.
						
			
			auto string_publisher_=this->create_publisher<std_msgs::msg::String>("string_topic",5);
			//params are period, callback, group
			auto timer_=this->create_wall_timer(
					std::chrono::seconds(1), //every second
					//std::bind(a,b) makes it so that function a is called from b (maybe??)
					std::bind(&IntroPublisher::timer_callback, this)
					);:
		}
		


	private:
		void timer_callback(){
			auto msg = std_msgs::msg::String();
			msg.data = "Hello " + std::to_string(count_++);
			RCLCPP_INFO(this->get_logger(),"Publishing: %s", msg.data.c_str());
			publisher_->publish(msg);
		}
		rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_; // Publisher
		rclcpp::TimerBase::SharedPtr timer_;
		int count_;


};
	



int main(int argc, char **argv){
	rclcpp::init(argc, argv);
	auto node = std::make_shared<IntroPublisher>(); 
	rclcpp::spin(node);
	rclcpp::shutdown();
	return 0;
}
